base/set.jl

```diff
@@ -103,6 +116,33 @@ const ⊆ = issubset
 ⊊(l::Set, r::Set) = <(l, r)
 ⊈(l::Set, r::Set) = !⊆(l, r)
 
+function firstnunique(C, n::Integer)
+    out = Vector{eltype(C)}()
```

Comment:
worth `sizehint!`-ing `out` (and maybe `seen` if `sizehint!` is useful for `Set`s) to expect `n` elements?